patches/minecraft/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java.patch

```diff
@@ -1,5 +1,16 @@
 --- a/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
 +++ b/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+@@ -150,7 +_,10 @@
+       return adouble;
+    }
+ 
++   @javax.annotation.Nullable private Player unlimitedLastHurtByPlayer = null;
+    public void m_8107_() {
++      if (this.f_20888_ != null) this.unlimitedLastHurtByPlayer = f_20888_;
++      if (this.unlimitedLastHurtByPlayer != null && this.unlimitedLastHurtByPlayer.m_146910_()) this.unlimitedLastHurtByPlayer = null;
```

Comment:
Could you maybe add a comment here explaining why we need to capture the player here? I know you already did this in the PR desc, but for future reference a code comment wouldn't hurt.